python/dgl/convert.py

```diff
@@ -429,94 +244,236 @@ def hetero_from_shared_memory(name):
 
 def heterograph(data_dict,
                 num_nodes_dict=None,
-                validate=True,
-                formats=['coo', 'csr', 'csc'],
                 idtype=None,
                 device=None):
-    """Create a heterogeneous graph from a dictionary between edge types and edge lists.
+    """Create a heterogeneous graph.
 
     Parameters
     ----------
-    data_dict : dict
-        The dictionary between edge types and edge list data.
+    data_dict : graph data
+        The dictionary data for constructing a heterogeneous graph. The keys are in the form of
+        string triplet (src_type, edge_type, dst_type), specifying the source node,
+        edge, and destination node types. The values are graph data in the form of
+        :math:`(U, V)`, where :math:`(U[i], V[i])` forms an edge and is given edge ID :math:`i`.
+        The allowed graph data formats are:
+
+        - ``(Tensor, Tensor)``: Each tensor must be a 1D tensor containing node IDs. DGL calls
+          this format "tuple of node-tensors". The tensors should have the same data type of
+          int32/int64 and device context (see below the descriptions of :attr:`idtype` and
+          :attr:`device`).
+        - ``(iterable[int], iterable[int])``: Similar to the tuple of node-tensors
+          format, but stores node IDs in two sequences (e.g. list, tuple, numpy.ndarray).
+    num_nodes_dict : dict[str, int], optional
+        The number of nodes for each node type. If not given (default), for each node type
+        :math:`T`, DGL finds the largest ID appeared in *every* graph data whose source or
+        destination node type is :math:`T`, and sets the number of nodes to be that ID plus one.
+        If given and the value is no greater than the largest ID, DGL will raise an error.
+    idtype : int32 or int64, optional
+        The data type for storing the structure-related graph information such as node and
+        edge IDs. It should be a framework-specific data type object (e.g., ``torch.int32``).
+        If ``None`` (default), DGL infers the ID type from the :attr:`data_dict` argument.
+    device : device context, optional
+        The device of the returned graph, which should be a framework-specific device object
+        (e.g., ``torch.device``). If ``None`` (default), DGL uses the device of the tensors of
+        the :attr:`data` argument. If :attr:`data` is not a tuple of node-tensors, the
```

Comment:
What will happen if device of different data is different.

Reply:
I mentioned in the description of ``data_dict`` that if it is a tuple of two tensors, they should have the same device.